workflow: l4lb: pass correct path for PR checkout

jibi opened the pull request:
Passing ${{ github.workspace }}/pull-request to the init.sh script as
path of the PR checkout is incorrect, as inside the VM the repository
will be under a different path.

Instead of that, pass the correct absolute path for the PR checkout.

Moreover, checkout first the upstream repo (under cilium/cilium) and
only after that the PR one (under cilium/cilium/pull-request), as
otherwise the upstream checkout will nuke the directory with PR one.

Fixes: #19953

nbusseneau reviewed (approved):
I would suggest we pass in the whole path (`../../pull-request/install/kubernetes/cilium`) as a proper `HELM_CHART_DIR` arg to the script, to help make the script and execution context independents, but this works as-is.

nbusseneau commented on test/l4lb/test.sh:
For backward compatibility: 
```suggestion
HELM_CHART_DIR=${3:-/vagrant/install/kubernetes/cilium}
```

nbusseneau commented on test/nat46x64/test.sh:
Likewise.

nbusseneau reviewed (approved):
LGTM, one nit.

jibi committed:
workflow: l4lb: pass correct path for PR checkout

Passing ${{ github.workspace }}/pull-request to the init.sh script as
path of the PR checkout is incorrect, as inside the VM the repository
will be under a different path.

Instead of that, pass the correct absolute path for the PR checkout.

Moreover, checkout first the upstream repo (under cilium/cilium) and
only after that the PR one (under cilium/cilium/pull-request), as
otherwise the upstream checkout will nuke the directory with PR one.

Fixes: #19953
Signed-off-by: Gilberto Bertin <jibi@cilium.io>
